IPython/core/display.py

```diff
@@ -199,6 +202,10 @@ def display(*objs, include=None, exclude=None, metadata=None, transient=None, di
         format = InteractiveShell.instance().display_formatter.format
 
     for obj in objs:
+        if isinstance(obj, DisplayObject) and obj.metadata:
+            temp_dict = deepcopy(obj.metadata)
+            temp_dict.update(metadata)
+            metadata.update(temp_dict)
```

Comment:
This seems awkward. I'm guessing it's part of the compatibility thing you mentioned you worked out with Min - if so, can we have a comment explaining why it's necessary?